### PR TITLE
provider/aws: Fixed the aws_vpc enable_dns_support handling on creation

### PR DESCRIPTION
**Pull request by Ninir**
Fixes https://github.com/hashicorp/terraform/issues/10168

This is an attempt in fixing the related issue. Still experimenting Go & Terraform codebase, be kind 😅

### CODE REVIEW

File: builtin/providers/aws/resource_aws_vpc.go [hunk 1]
```diff
@@ -260,7 +260,9 @@ func resourceAwsVpcUpdate(d *schema.ResourceData, meta interface{}) error {
 		d.SetPartial("enable_dns_support")
 	}
 
-	if d.HasChange("enable_dns_support") {
+	_, hasEnableDnsSupportOption := d.GetOk("enable_dns_support")
+
+	if !hasEnableDnsSupportOption || d.HasChange("enable_dns_support") {
```
Comment (Ninir) [inline builtin/providers/aws/resource_aws_vpc.go]: In fact, this will be the same for every attributes in this function, right?

**Comment by mattmoyer** [comment]
I confirmed that this does fix the buggy behavior, but I'm not sure I understand how.

It's worth noting that I _don't_ see the buggy behavior with `enable_dns_hostnames`, I think because it defaults to `false` on new VPCs whereas `enable_dns_support` defaults to `true`.

I don't 100% understand the bug but I suspect it's related to the initial default value not being properly reflected in the initial resource data structure when it's first passed to `resourceAwsVpcUpdate`.

**Comment by Ninir** [comment]
@mattmoyer In fact it does the following thing from the code POV:

1. It [creates the VPC](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_vpc.go#L95) without the option (it is how the API is designed).
2. Since there are no modifications on the given attribute, it is not written to the state.
3. After creating the resource, it [updates the resource](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_vpc.go#L131) (in order to add attributes).
4. Since [it does not have the attribute in the state](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_vpc.go#L263), this line resolves to false, because the value has not changed.
5. When you apply again, [Terraform will read the value from AWS](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_vpc.go#L167), and will set the retrieved value (true)
6. Since the value is different from your config one, it will warn about the update

Hope it explains better what happens!

File: builtin/providers/aws/resource_aws_vpc.go (hunk 1 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_vpc.go]: seems `enable_dns_support` is the only attribute that defaults `true` on AWS side here, which is why this is the only one that needs this check (if I understand correctly)

**Pull request merged**